src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error';
+import { SortTypes } from './types/SortTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [query, setQuery] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState('all');
+  const [hasError, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    async function loadTodos() {
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+      } catch (error) {
+        setError('An error occurred while loading the goods.');
+      }
+    }
+
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    let searchedTodos = [...todos];
```

Comment:
Method `filter` returns a new array every time, so you don’t need to copy it with the spread operator.